### DISCUSSION TITLE
[Announcement] Migration to ASP.NET Core (Obsoleting ASP.NET MVC / .NET Framework)

### DISCUSSION THREAD

**Discussion by volkanceylan** [serenity-is/Serenity#5738] [Announcements]
Hello everyone,

As you are probably aware ASP.NET MVC did not get any updates for years, and is an obsolete technology. 

ASP.NET Core 1 and 2 used to support both .NET Core and .NET Framework but Microsoft removed the support in .NET Core 3, so .NET Framework itself is also obsolete.

.NET 5 will be the next single .NET and it will be based on .NET Core. (They are simply skipping version 4 to avoid confusion with .NET Framework 4).

We tried to support both ASP.NET MVC and ASP.NET Core for a long time, but it has been a hard work and resulted in some code in ASP.NET Core version to be not so .NET Core'ish, like Dependency system (we should have used .NET Core dependency injection container but had to use some workarounds to keep compatibility with MVC version).

We are planning not to release any new ASP.NET MVC versions of Serene / StartSharp unless there will be a critical issue that needs to be solved. 

> For existing StartSharp customers, we'll provide support on how to apply new features in StartSharp Core to their StartSharp MVC based project, if they can't migrate to .NET Core for some reason, until their subscription ends.

We'll be still releasing .NET Framework versions of Serenity base libraries until end of 2020, so you may still update Serenity itself with the package manager.

We strongly recommend migrating your existing MVC projects (that needs to be updated) to .NET Core version and prepared a migration guide for you:

https://serenity.is/blog/2020/06/17/migrating-dotnetcore

The steps in above document is tested with several projects, but it is still possible that you may have different issues due to custom code in your projects, or if your project is based on a relatively older version of Serene/StartSharp.

If you report your conversion issues here, we may offer solutions when possible, and may add your solutions to the document itself.

**Comment by stixoffire** [comment]
@volkanceylan  Thank you for the well written information providing reasons for the change .this will help me in my next iteration to migrate with more time allotted ..
